### PR TITLE
move chat events

### PR DESCRIPTION
**Pull request by flexsurfer**
image

### CODE REVIEW

File: src/status_im2/contexts/chat/events.cljs [hunk 1]
```diff
@@ -0,0 +1,324 @@
+(ns status-im2.contexts.chat.events
```
Comment (flexsurfer) [inline src/status_im2/contexts/chat/events.cljs:1]: HERE

**Comment by flexsurfer** [comment]
idk tests are broken, i have no idea why
1) there is no root subscriptions for subscriptions tests for example `status-im2.subs.wallet.wallet-test`
but how has it worked before?
2) 
```
ERROR in (open-activity-center-test) (TypeError:NaN:NaN)
Uncaught exception, not in assertion.
expected: nil
  actual: #object[TypeError TypeError: Cannot read properties of null (reading 'call')]
```
3)
```
ERROR in (open-activity-center-test) (ReferenceError:NaN:NaN)
Uncaught exception, not in assertion.
expected: nil
  actual: #object[ReferenceError ReferenceError: day8 is not defined]

```
4)`status-im.integration-test` stuck 
```
DEBUG [utils.re-frame:18] - Handling re-frame event:  :signals/signal-received
signal received {"type":"discovery.stopped","event":null}
DEBUG [utils.re-frame:18] - Handling re-frame event:  :signals/signal-received
DEBUG [status-im.signals.core:114] - Event  discovery.stopped  not handled
INFO [01-19|15:58:55.704] Messenger failed to process sent messages error="Can't get raw message with id 0xd7ee57d6402b80c625347dabe0e7129913e08b784156a6f079ae3f7d799db36f: sql: no rows in result set"
INFO [01-19|15:58:56.485] processing message                       site=RetrieveAll hash=0x03ac689b040182faf27b8f71271eb38c9af14816240ca4fa08f23a595d7573fd message-id=0xd7ee57d6402b80c625347dabe0e7129913e08b784156a6f079ae3f7d799db36f
ERROR[01-19|15:58:58.501] could not retrieve message               error="record not found"
WARN [01-19|15:58:58.501] message not retrieved 
INFO [01-19|15:59:20.484] publishing records                       records=[]
INFO [01-19|15:59:20.485] received records                         records=[]
INFO [01-19|15:59:50.484] publishing records                       records=[]
INFO [01-19|15:59:50.484] received records                         records=[]
INFO [01-19|16:00:20.482] publishing records                       records=[]
INFO [01-19|16:00:20.482] received records                         records=[]
INFO [01-19|16:00:50.484] publishing records                       records=[]
INFO [01-19|16:00:50.484] received records                         records=[]
INFO [01-19|16:00:50.605] Messenger failed to process sent messages error="Can't get raw message with id 0x2a74a302df6fd24d2d27dda967d554d976309f036eaee241af9db20acda21a34: sql: no rows in result set"
INFO [01-19|16:00:51.485] processing message                       site=RetrieveAll hash=0x3c0e3bac17d47048c59d4089d1a28bf1cca490fb42f4ed7343f8097a2ecba1b4 message-id=0x2a74a302df6fd24d2d27dda967d554d976309f036eaee241af9db20acda21a34
ERROR[01-19|16:00:53.502] could not retrieve message               error="record not found"
WARN [01-19|16:00:53.502] message not retrieved 
INFO [01-19|16:01:20.483] publishing records                       records=[]
INFO [01-19|16:01:20.483] received records                         records=[]
```

@ilmotta @J-Son89 pls help

**Comment by ilmotta** [comment]
> 

Hey @flexsurfer, I'm gonna take a look now.

**Comment by J-Son89** [comment]
> > 
> 
> 
> 
> Hey @flexsurfer, I'm gonna take a look now.

Nice one, I'm not so familiar with activity centre but can take a look if needs be 😀

**Comment by flexsurfer** [comment]
hey @ilmotta i think i found the reason for 4, need time to see if it fixes other issues somehow

**Comment by flexsurfer** [comment]
yes, so 1-3 happens if remove `status-im.integration-test` thats weird but probably that ok? 
4 happens because there is no registered handler for
(rf-test/wait-for
               [::chat.models/mute-chat-toggled-successfully]
we need something to catch this, like throw an error if there is not such event registered or have a timeout with error

**Comment by ilmotta** [comment]
@flexsurfer:

>there is no root subscriptions for subscriptions tests for example status-im2.subs.wallet.wallet-test
but how has it worked before?

The way to go here is for subscription tests to only require the root namespace and not the individual subscription file. This way, the entire graph of subscriptions is registered correctly during test execution. I don't know why I haven't done that in the past...

There are only a handful of subscription test files, and it'd be great if you could change the namespaces to require `status-im2.subs.root`, just like the integration test file does.

------

```
ERROR in (open-activity-center-test) (TypeError:NaN:NaN)
Uncaught exception, not in assertion.
expected: nil
  actual: #object[TypeError TypeError: Cannot read properties of null (reading 'call')]
```

If I ignore the integration tests, all unit tests are passing. I double-checked in the REPL too, and it's good. This test should always pass, so something else is going on.

I can't reproduce this error locally.

-----

> status-im.integration-test stuck

I believe this is due to the fact that some integration tests are waiting for certain events to be dispatched (see there are many calls to `rf-test/wait-for`), but they are not happening anymore, clearly because this PR changed something that affected them.

I can try to investigate this further, but just a heads up, this could take a long time.

**Comment by flexsurfer** [comment]
hey @ilmotta no need to spend time, see my comments above

**Comment by ilmotta** [comment]
> we need something to catch this, like throw an error if there is not such event registered or have a timeout with error

@flexsurfer I thought the library `day8.re-frame.test` had a default reasonable timeout with a proper error message. If not, then yeah, this should improve

**Comment by flexsurfer** [comment]
tests are useless and harmful, i spent 2h just fixing tests

**Comment by flexsurfer** [comment]
could someone approve pls, @ilmotta @J-Son89 @Parveshdhull

**Review by ilmotta — APPROVED** [review]

**Comment by ilmotta** [comment]
> tests are useless and harmful, i spent 2h just fixing tests

Which ones are you referring to @flexsurfer? We shouldn't have useless tests in the test suite, only valuable ones.

**Comment by flexsurfer** [comment]
all :(

**Comment by ilmotta** [comment]
> all :(

That's kind of harsh @flexsurfer ;) Jokes aside, sounds like a topic to be calmly discussed with the team, because there are many nuances about how to create a good test suite.

**Comment by flexsurfer** [comment]
it's just my opinion, and its not popular, i guess im the only one, but for six years i've never seen a case when a test found a bug, but i seen a lot of work just to fix broken tests

**Comment by ilmotta** [comment]
> it's just my opinion, and its not popular, i guess im the only one, but for six years i've never seen a case when a test found a bug, but i seen a lot of work just to fix broken tests

Well it's a long discussion @flexsurfer. But fair enough, I think you had past experiences that gave you only one side of the coin if you know what I mean. I've seen both cases many times, test suites with little value and rock star level test suites that allowed teams to do continuous delivery without costly and slow processes.

The way we deliver software in Status Mobile is very painfully SLOW compared to what it could be. And this is highly associated with the lack of automated tests that developers own, manage and design. And I'm not talking about unit tests alone.

Your opinion is very important, thanks. I'll keep this in the back of my mind.

**Comment by flexsurfer** [comment]
yeah @ilmotta  but we have 3 points:
1) we have e2e and QA
2) clojurescript and re-frame - this one is important
3) we're not in a hurry

i agree unit tests are useful sometimes, probably would be cool to have performance tests for subs and events, but having integration tests have no sense because 1 and 2 
our app is pretty simple and modular, so its really hard to break it, and also tests always teach developers to be less mindful of the new code they write, i feel like its better to have devs who care about code and could check places they can break and simple manual testing instead of writing tests and rely on them

for 1 it seems like we do double work, because anyway all will be covered by e2e and QA, yeah for sure it's better to find the bug on early stage, but as i said i've never seen such cases, usually real bugs still found only on QA stage

**Comment by ilmotta** [comment]
Hey @flexsurfer, 

I think you're touching on some important topics, like different types of tests. Every type of test has a purpose, can catch different classes of bugs, have different performance trade-offs, can be more/less convenient do pinpoint the test failure, can be flaky/deterministic, and so on and so forth.

Also interesting is the application's architecture, as you said re-frame really shines in helping the frontend be more manageable, although it's just one piece of the game, it's a strong piece nonetheless.

I appreciate the fact that we can all respectfully disagree, even when our opinions are at opposite ends of the spectrum regarding tests :)

There are so many different things to discuss about testing that it would be unproductive for us to go into this rabbit hole over this PR, but count me in for such discussions in a more appropriate moment/place.

**Pull request merged**